nixos/modules/services/misc/redmine.nix

```
@@ -395,9 +395,13 @@ in
 
 
         # handle database.passwordFile & permissions
-        DBPASS=${optionalString (cfg.database.passwordFile != null) "$(head -n1 ${cfg.database.passwordFile})"}
         cp -f ${databaseYml} "${cfg.stateDir}/config/database.yml"
-        sed -e "s,#dbpass#,$DBPASS,g" -i "${cfg.stateDir}/config/database.yml"
+
+        ${optionalString (cfg.database.passwordFile != null) ''
+          DBPASS="$(head -n1 ${cfg.database.passwordFile})"
+          sed -e "s,#dbpass#,$DBPASS,g" -i "${cfg.stateDir}/config/database.yml"
```

You could replace `sed` with `replace-secret` to prevent errors with passwords containing `,` and other weird characters. Also it prevents the secret from being leaked through `/proc/<pid>/cmdline`.

```suggestion
          chmod u+w '${cfg.stateDir}/config/database.yml'
          ${lib.getExe pkgs.replace-secret} '#dbpass#' '${cfg.database.passwordFile}' '${cfg.stateDir}/config/database.yml'
```

---

Thanks for the suggestion. Will have a look at some point later. It's not something I introduce, so I'm closing this for now.